client/src/main/java/io/pravega/client/stream/impl/EventStreamReaderImpl.java

```diff
@@ -74,12 +81,18 @@
     @GuardedBy("readers")
     private final List<EventSegmentReader> readers = new ArrayList<>();
     @GuardedBy("readers")
-    private final Map<Segment, Range> ranges = new HashMap<>();
+    private Map<Segment, Range> ranges = new HashMap<>();
     @GuardedBy("readers")
     private final Map<Segment, Long> sealedSegments = new HashMap<>();
     @GuardedBy("readers")
     private Sequence lastRead;
     @GuardedBy("readers")
+    private Map<Segment, Long> ownedSegments = new HashMap<>();
+    @GuardedBy("readers")
```

Comment:
One of the approaches taken by this PR is to use as input for building `PositionImpl` objects references to the collections they need to build their state instead of copies (specially map copies are a main performance bottleneck and a source of GC activity). While we do not make unsafe accesses or manipulations from `PositionImpl` objects to these shared collections, multiple `PositionImpl` object may point to `ownedSegments`, `ranges` and `segmentOffsetUpdates`. Shall I remove `@GuardedBy("readers")` from these three collections? 
(Note that start adding locks here would impact on performance, so I prefer not to use additional synchronization. In my view, the current code has been reviewed and intensively benchmarked to be sure that is thread safe).

Reply:
I have removed the annotation from the collections that are passed as references to the `PositionImpl` builder.